internal/controller/services/monitoring/monitoring_controller_actions.go

```diff
@@ -49,10 +53,24 @@ func initialize(_ context.Context, rr *odhtypes.ReconciliationRequest) error {
 func updatePrometheusConfigMap(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
 	// Map component names to their rule prefixes
 	dsc, err := cluster.GetDSC(ctx, rr.Client)
+	// If the DSC doesn't exist, we don't need to update the prometheus configmap
+	if err != nil && k8serr.IsNotFound(err) {
+		return nil
+	}
 	if err != nil {
 		return fmt.Errorf("failed to get DataScienceCluster instance: %w", err)
 	}
 
+	dsci, err := cluster.GetDSCI(ctx, rr.Client)
+	if err != nil {
+		return fmt.Errorf("failed to get DataScienceInitialisation instance: %w", err)
+	}
+
+	// If monitoring is unmanaged or the release is not managed, we don't need to update the prometheus configmap
+	if dsci.Spec.Monitoring.ManagementState == operatorv1.Unmanaged || rr.Release.Name != cluster.ManagedRhoai {
+		return nil
+	}
```

Comment:
Wouldn't this now disallow prometheus config map updates on self-managed cluster with monitoring set to managed?

Reply:
Yes. There is a follow on JIRA to allow turning on alerting on self-managed and ODH, at that point we will be managing the prometheus config.